ggshield/core/git_hooks/ci/previous_commit.py

```diff
@@ -282,6 +271,21 @@ def azure_pull_request_previous_commit_sha(verbose: bool) -> Optional[str]:
     return None
 
 
+def get_new_branch_before_sha(branch_name: str, verbose: bool) -> str:
+    new_commits = get_new_branch_ci_commits(branch_name, Path.cwd(), "origin")
+    if new_commits:
+        new_branch_before_sha = f"{new_commits[-1]}^1"
+        if verbose:
+            click.echo(
+                f"new_branch_before_sha: {new_branch_before_sha}\n",
+                err=True,
+            )
+        return new_branch_before_sha
+    else:
+        # New branch pushed with no commits
+        return "HEAD"
```

Comment:
If I am not mistaken, this only works if `branch_name` is the current branch. I think it would be more robust to return `branch_name` here.

And maybe use an early return:

```python
    new_commits = get_new_branch_ci_commits(branch_name, Path.cwd(), "origin")
    if not new_commits:
        return branch_name  
    new_branch_before_sha = f"{new_commits[-1]}^1"
    if verbose:
        click.echo(
            f"new_branch_before_sha: {new_branch_before_sha}\n",
            err=True,
        )
    return (
        new_branch_before_sha
        if is_valid_git_commit_ref(new_branch_before_sha)
        else None
    )
```

Reply:
You're right. I've made the change with two differences:
- the branch_name returned is prefixed with "refs/remotes/origin/", as it seems this is how other branches should be accessed in CI environments
- If this happens to not resolve to a valid ref, the function falls back to HEAD instead